extensions/lifecycle/tasks/LifecycleDeleteObjectTask.js

```diff
@@ -86,19 +91,23 @@ class LifecycleDeleteObjectTask extends BackbeatTask {
         const bucket = entry.getAttribute('target.bucket');
         const key = entry.getAttribute('target.key');
         const lastModified = entry.getAttribute('details.lastModified');
-
         if (lastModified) {
             const reqParams = {
                 Bucket: bucket,
                 Key: key,
-                IfUnmodifiedSince: lastModified,
+                IfUnmodifiedSince: new Date(lastModified),
             };
-            const req = s3Client.headObject(reqParams);
-            attachReqUids(req, log);
-            return req.send((err, res) => {
-                LifecycleMetrics.onS3Request(log, 'headObject', 'expiration', err);
-                return done(err, res);
-            });
+            const command = new HeadObjectCommand(reqParams);
+            attachReqUids(command, log);
+            return s3Client.send(command)
+                .then(res => {
+                    LifecycleMetrics.onS3Request(log, 'headObject', 'expiration', null);
+                    return done(null, res);
+                })
+                .catch(err => {
+                    LifecycleMetrics.onS3Request(log, 'headObject', 'expiration', err);
+                    return done(err);
+                });
```

Comment:
dedup non-trivial code:
```suggestion
            const cb = (err, res) > {
                LifecycleMetrics.onS3Request(log, 'headObject', 'expiration', err);
                return done(err, res);
            }
            return s3Client.send(command).then(res => cb(null, res), res);
```